Add STM32N6 support

gautierg-st opened the pull request:
This PR adds the support for the STM32N6 series.
It adds support of two boards:
* STM32N6570 DK
* Nucleo N657x0-Q

This PR includes the support of GPIO and UART. Other peripherals will soon be added in future PRs

gautierg-st committed:
soc: st: stm32: add stm32n6 series

Add STM32N6 series

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
dts: bindings: clock: add stm32n6 rcc clocks

Add STM32N6 RCC clock bindings

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
drivers: clock: add stm32n6 clock

Add STM32N6 clock driver.
Supported clocks are LSE, LSI, HSE, HSI, PLL1-4, IC1-20,
peripheral clock and CPU clock.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
drivers: pinctrl: stm32: add pinctrl for stm32n6

Add pinctrl driver for STM32N6

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
drivers: gpio: stm32: add support for stm32n6

Add GPIO support for STM32N6

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
include: zephyr: dt-bindings: reset: add stm32n6

Add STM32N6 dt bindings for reset

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
dts: arm: st: n6: add dtsi for stm32n6 series

Add dtsi files for STM32N6 series

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
boards: st: stm32n6570_dk: add support for stm32n6570_dk board

Add support for STM32N657 DK board.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
west.yml: update west.yml for stm32n6 hal

Update west.yml for STM32N6 HAL

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
dts: arm: st: n6: add all u(s)art instances

Add U(S)ART2 to 10 to the device tree.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

erwango committed:
boards: st: Add support for the nucleo_n657x0_q

Add support for the nucleo_n657x0_q board.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

gautierg-st committed:
scripts: stm32cubeprogrammer: Add possibility to flash a bin file

Add the option to flash a bin file if one is available from the board's
cmake configuration.
Use of elf has prio over this, and bin has prio over hex which remains
the default option.
Also add a parameter to stm32cubeprogrammer runner to specify the download
address.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

pdgendt commented on drivers/clock_control/clock_stm32_ll_n6.c:
```diff
@@ -0,0 +1,957 @@
+	return 0;
+}
+
+static struct clock_control_driver_api stm32_clock_control_api = {
```
```suggestion
static DEVICE_API(clock_control, stm32_clock_control_api) = {
```

gautierg-st replied:
Done

erwango commented on scripts/west_commands/runners/stm32cubeprogrammer.py:
```diff
@@ -234,12 +246,23 @@ def flash(self, **kwargs) -> None:
 
         # flash image and run application
         dl_file = self.cfg.elf_file if self._use_elf else self.cfg.hex_file
+        if self._use_elf:
+            dl_file = self.cfg.elf_file
+            flash_and_run_args = ["--download", dl_file]
+        elif self.cfg.bin_file is not None and os.path.isfile(self.cfg.bin_file) and \
+            "zephyr_signed" in self.cfg.bin_file:
+            dl_file = self.cfg.bin_file
+            dl_address = f"0x{self._download_address:X}"
+            flash_and_run_args = ["--download", dl_file, dl_address]
+        elif self.cfg.hex_file is not None and os.path.isfile(self.cfg.hex_file):
+            # --user-elf not used and no bin file given, default to hex
+            dl_file = self.cfg.hex_file
+            flash_and_run_args = ["--download", dl_file]
         if dl_file is None:
             raise RuntimeError('cannot flash; no download file was specified')
         elif not os.path.isfile(dl_file):
             raise RuntimeError(f'download file {dl_file} does not exist')
```
Please adapt to the following to ease further changes for USB-DFU support

```suggestion
        if self._use_elf:
            dl_file = self.cfg.elf_file
        elif self.cfg.bin_file is not None and os.path.isfile(self.cfg.bin_file) and \
            "zephyr_signed" in self.cfg.bin_file:
            dl_file = self.cfg.bin_file
        elif self.cfg.hex_file is not None and os.path.isfile(self.cfg.hex_file):
            # --user-elf not used and no bin file given, default to hex
            dl_file = self.cfg.hex_file
        if dl_file is None:
            raise RuntimeError('cannot flash; no download file was specified')
        elif not os.path.isfile(dl_file):
            raise RuntimeError(f'download file {dl_file} does not exist')
            
        flash_and_run_args = ["--download", dl_file]
        if self._download_address is not None:
            flash_and_run_args.append(f"0x{self._download_address:X}")
```

nordicjm commented on boards/st/nucleo_n657x0_q/Kconfig.defconfig:
```diff
@@ -0,0 +1,10 @@
+
+if BOARD_NUCLEO_N657X0_Q
+
+# Nothing for now.
```
file to go

nordicjm commented on boards/st/nucleo_n657x0_q/board.cmake:
```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2024 STMicroelectronics
+
+if(${CMAKE_HOST_SYSTEM_NAME} STREQUAL Windows)
+    set(SIGNING_TOOL STM32_SigningTool_CLI.exe)
```
2 space indent

gautierg-st replied:
Done

nordicjm commented on boards/st/nucleo_n657x0_q/board.cmake:
```diff
@@ -0,0 +1,26 @@
+    set(SIGNING_TOOL STM32_SigningTool_CLI)
+endif()
+
+set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
+  COMMAND ${SIGNING_TOOL}
+  -bin ${PROJECT_BINARY_DIR}/${CONFIG_KERNEL_BIN_NAME}.bin
+  -nk -of 0x80000000 -t fsbl -hv 2.3
+  -o ${PROJECT_BINARY_DIR}/${CONFIG_KERNEL_BIN_NAME}_signed.bin
+  -dump ${PROJECT_BINARY_DIR}/${CONFIG_KERNEL_BIN_NAME}_signed.bin
+  WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+)
+
+set_property(TARGET runners_yaml_props_target PROPERTY bin_file zephyr_signed.bin)
```
what is this and why is this here? This should not be in a board

gautierg-st replied:
STM32N6 requires binaries to be signed before flashing them. Otherwise they won't be executed.
This code is used to call the STM32_SigningTool_CLI utility to sign the binary as the target expects it.
This operation is necessary but if its place is elsewhere we can move it.

nordicjm replied:
this should be a soc thing, and it doesn't sign the hex file so that's wrong plus will completely fail if bin output is disabled in Kconfig. Would also be nice to match mcuboot signing name i.e. `<kernel>.signed.bin`

gautierg-st replied:
Moving it to soc and renaming the signed binary works, but unfortunately we have no choice but to sign the bin file because the signing tool requires it:
`Error: binary file is not .bin or .stm32 extension`

gautierg-st replied:
Moved the code to soc and renamed to <kernel>.signed.bin. But still using bin file because we have to.

nordicjm commented on boards/st/nucleo_n657x0_q/doc/index.rst:
```diff
@@ -0,0 +1,166 @@
+Supported Features
+==================
+
+The Zephyr nucleo_n657x0_q board configuration supports the following hardware features:
```
```suggestion
The Zephyr ``nucleo_n657x0_q`` board supports the following hardware features:
```

gautierg-st replied:
Done

nordicjm commented on boards/st/nucleo_n657x0_q/doc/index.rst:
```diff
@@ -0,0 +1,166 @@
+Debugging
+=========
+
+
```
missing text?

gautierg-st replied:
Completed

nordicjm commented on boards/st/nucleo_n657x0_q/nucleo_n657x0_q_defconfig:
```diff
@@ -0,0 +1,17 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2024 STMicroelectronics
+
+# enable uart driver
```
capitalise first letters and UART not uart

gautierg-st replied:
Done

nordicjm commented on boards/st/stm32n6570_dk/Kconfig.defconfig:
```diff
@@ -0,0 +1,10 @@
+# Copyright (c) 2024 STMicroelectronics
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_STM32N6570_DK
```
all comments in other commit apply here too

gautierg-st replied:
Done

gautierg-st committed:
soc: st: stm32: stm32n6x: add signing tool

Generate signed binary necessary to start a program from Flash on STM32N6

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

nordicjm commented on drivers/clock_control/clock_stm32_ll_n6.c:
```diff
@@ -222,9 +227,14 @@ static inline int stm32_clock_control_off(const struct device *dev,
 		       pclken->enr);
 
+	/* Clear Low Power clock */
+	sys_clear_bits(DT_REG_ADDR(DT_NODELABEL(rcc)) + pclken->bus + STM32_CLOCK_LP_BUS_SHIFT,
```
commit needs to be squashed with previous commit

nordicjm commented on drivers/clock_control/clock_stm32_ll_n6.c:
```diff
@@ -477,6 +477,10 @@ static void stm32_clock_switch_to_hsi(void)
 	while (LL_RCC_GetSysClkSource() != LL_RCC_SYS_CLKSOURCE_STATUS_HSI) {
 	}
+
+	LL_RCC_SetCpuClkSource(LL_RCC_CPU_CLKSOURCE_HSI);
```
and this commit

gautierg-st replied:
Done

nordicjm commented on dts/arm/st/n6/stm32n6.dtsi:
```diff
@@ -0,0 +1,378 @@
+			interrupts = <159 0>;
+			status = "disabled";
+		};
+
```
```suggestion
```

nordicjm replied:
not addressed

nordicjm commented on dts/arm/st/n6/stm32n657X0.dtsi:
```diff
@@ -0,0 +1,16 @@
+
+	axisram2: memory@34180400 {
+		reg = <0x34180400 DT_SIZE_K(511)>;
+	};};
```
missing newline

gautierg-st replied:
Done

nordicjm commented on soc/st/stm32/stm32n6x/CMakeLists.txt:
```diff
@@ -8,3 +8,20 @@ zephyr_sources(
+set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
+  COMMAND ${SIGNING_TOOL}
+  -in ${PROJECT_BINARY_DIR}/${CONFIG_KERNEL_BIN_NAME}.bin
+  -nk -of 0x80000000 -t fsbl -hv 2.3
```
address should come from dts using cmake functions

gautierg-st replied:
-of is for option flags. It is in fact not needed -> removed.

nordicjm commented on soc/st/stm32/stm32n6x/CMakeLists.txt:
```diff
@@ -8,3 +8,20 @@ zephyr_sources(
+  COMMAND ${SIGNING_TOOL}
+  -in ${PROJECT_BINARY_DIR}/${CONFIG_KERNEL_BIN_NAME}.bin
+  -nk -of 0x80000000 -t fsbl -hv 2.3
+  -o ${PROJECT_BINARY_DIR}/${CONFIG_KERNEL_BIN_NAME}.signed.bin
```
also the soc in `Kconfig` needs to select the build output bin Kconfig, otherwise the board is unusable

erwango commented on soc/st/stm32/stm32n6x/soc.c:
```diff
@@ -0,0 +1,67 @@
+#if defined(CONFIG_SOC_RESET_HOOK)
+void soc_reset_hook(void)
+{
+	/* This is provided by STM32Cube HAL */
+	SystemInit();
```
Can we replace directly by the following to minimize dependencies ?
```
  SCB->VTOR = INTVECT_START;
```
(As I guess `USER_VECT_TAB_ADDRESS` is not defined).

And btw, isn't it already set as part of zephr arch init ?

gautierg-st replied:
SystemInit contains a lot more initialization than just this line :)

erwango replied:
ok, indeed, my bad. Let's keep it that way then.

nordicjm commented on boards/st/stm32n6570_dk/stm32n6570_dk.dts:
```diff
@@ -0,0 +1,7 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "stm32n6570_dk_common.dtsi"
```
bit confused why there is a dts including a board specific dtsi when there is no other dts file for this board?

gautierg-st replied:
It's a provision for later.
The current dts is the standard one, using external flash. After flashing, a boot pin needs to be switched to start the program which poses problem for automatic testing.
We'll add another dts that will use serial boot. It doesn't require switching the boot pin and we will use that on our test bench.
I didn't add it in this PR to try to keep things minimal.

nordicjm replied:
well test files should not be here, you can use board extensions for that https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#board-extensions and have your test files in your own testing repo

gautierg-st replied:
Common dtsi file removed. Only one dts remains for each board.

nordicjm commented on boards/st/stm32n6570_dk/stm32n6570_dk_common.dtsi:
```diff
@@ -0,0 +1,109 @@
+
+	leds: leds {
+		compatible = "gpio-leds";
+		green_led_1: led_1 {
```
inconsistent spacing in this file

nordicjm commented on boards/st/stm32n6570_dk/stm32n6570_dk_common.dtsi:
```diff
@@ -0,0 +1,109 @@
+			label = "User LD2";
+		};
+	};
+
```
```suggestion
```

nordicjm commented on boards/st/stm32n6570_dk/stm32n6570_dk_common.dtsi:
```diff
@@ -0,0 +1,109 @@
+	model = "STMicroelectronics STM32N6570 Discovery Kit board";
+	compatible = "st,stm32n6570-dk";
+
+	#address-cells = <1>;
+	#size-cells = <1>;
```
why are these needed?

gautierg-st replied:
Not needed -> removed

nordicjm commented on boards/st/nucleo_n657x0_q/nucleo_n657x0_common.dtsi:
```diff
@@ -0,0 +1,117 @@
+
+	leds: leds {
+		compatible = "gpio-leds";
+		green_led: led_1 {
```
inconsistent spacing

nordicjm commented on boards/st/nucleo_n657x0_q/nucleo_n657x0_q.dts:
```diff
@@ -0,0 +1,7 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "nucleo_n657x0_common.dtsi"
```
as other comment

erwango commented on drivers/clock_control/clock_stm32_ll_n6.c:
```diff
@@ -0,0 +1,933 @@
+	default:
+		return -ENOTSUP;
+	}
+
```
Missing update wrt https://github.com/zephyrproject-rtos/zephyr/pull/82540/commits/82f659ec86949974efc9ff770010b56b2dfdcf7f

gautierg-st replied:
Done

erwango commented on boards/st/stm32n6570_dk/stm32n6570_dk_common.dtsi:
```diff
@@ -0,0 +1,109 @@
+};
+
+&usart1 {
+	clocks = <&rcc STM32_CLOCK_BUS_APB2 0x00000010>,
```
```suggestion
	clocks = <&rcc STM32_CLOCK(APB2, 4)>,
```

Apply everywhere

gautierg-st replied:
Done

erwango commented on boards/st/stm32n6570_dk/stm32n6570_dk_defconfig:
```diff
@@ -0,0 +1,16 @@
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+# For now store the program in RAM during flashing
```
Outdated. Aslo, should it be part of SoC config ?

gautierg-st replied:
Keeping CONFIG_XIP=n. It gets overwritten if set in the SoC.
Removed CONFIG_FLASH_SIZE and CONFIG_FLASH_BASE_ADDRESS as they are automatically set to 0 when XIP=n

gautierg-st commented on scripts/west_commands/tests/test_stm32cubeprogrammer.py:
```diff
@@ -355,6 +367,7 @@
         "port": "swd",
         "frequency": None,
         "reset_mode": None,
+        "download_address": None,
```
Done

etienne-lms commented on include/zephyr/dt-bindings/clock/stm32n6_clock.h:
```diff
@@ -0,0 +1,193 @@
+
+/** @brief Device domain clocks selection helpers */
+/** CCIPR1 devices */
+#define ADF1_SEL(val)		STM32_DOMAIN_CLOCK(val, 7, 0, CCIPR1_REG)
```
Add parentheses around `val` here and in  lines below.

gautierg-st replied:
Done

etienne-lms commented on include/zephyr/dt-bindings/clock/stm32n6_clock.h:
```diff
@@ -0,0 +1,193 @@
+#define ICxCFGR_REG(ic)		(0xC4 + (ic - 1) * 4)
+
+/** @brief Divider ICx source selection */
+#define ICx_PLLy_SEL(ic, pll)	STM32_DOMAIN_CLOCK(pll-1, 3, 28, ICxCFGR_REG(ic))
```
Add parenthese around `pll` (+ space chars around `-`?)

etienne-lms commented on include/zephyr/dt-bindings/clock/stm32n6_clock.h:
```diff
@@ -0,0 +1,193 @@
+#define CFGR1_REG		0x20
+
+/** @brief CPU clock switch selection */
+#define CPU_SEL(val)		STM32_DOMAIN_CLOCK(val, 3, 16, CFGR1_REG)
```
ditto for `val`

gautierg-st replied:
Done

etienne-lms commented on include/zephyr/dt-bindings/clock/stm32n6_clock.h:
```diff
@@ -0,0 +1,193 @@
+#define LPUART1_SEL(val)	STM32_DOMAIN_CLOCK(val, 7, 8, CCIPR14_REG)
+
+/** @brief RCC_ICxCFGR register offset (RM0468.pdf) */
+#define ICxCFGR_REG(ic)		(0xC4 + (ic - 1) * 4)
```
Ditto for  #`ìc`

gautierg-st replied:
Done

etienne-lms reviewed (commented):
Comments for commit "drivers: clock: add stm32n6 clock".

erwango commented:
@nordicjm PTAL, also would you mind closing the resolved comment ?

gautierg-st committed:
modules: stm32: add stm32n6 cube modules

Add STM32N6 Cube HAL and LL modules

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

etienne-lms commented on include/zephyr/dt-bindings/clock/stm32n6_clock.h:
```diff
@@ -0,0 +1,193 @@
+ * @param reg RCC_CCIPRx register offset
+ * @param shift Position within RCC_CCIPRx.
+ * @param mask Mask for the RCC_CCIPRx field.
+ * @param val Clock value (0, 1, ... 7).
```
Order: reverse inline description comment lines?

gautierg-st replied:
It is done in this order on all series, so to be consistent, it would have to be changed everywhere. I think it can be done later if it is deemed necessary.

etienne-lms commented on boards/st/stm32n6570_dk/stm32n6570_dk.dts:
```diff
@@ -0,0 +1,105 @@
+
+	leds: leds {
+		compatible = "gpio-leds";
+		green_led_1: led_1 {
```
Still need an empty line above this line and line 32.

gautierg-st replied:
Done

etienne-lms commented on boards/st/nucleo_n657x0_q/nucleo_n657x0_q.dts:
```diff
@@ -0,0 +1,119 @@
+
+	leds: leds {
+		compatible = "gpio-leds";
+		green_led: led_1 {
```
Still need an empty line above this node and the 2 nodes below.
Ditto at line 40.

gautierg-st replied:
Done

etienne-lms commented on modules/Kconfig.stm32:
```diff
@@ -656,7 +691,7 @@ config USE_STM32_LL_DELAYBLOCK
 	bool
 	help
-	  Enable STM32Cube DelayBlock (DELAYBLOCK) LL module driver (stm32U5)
+	  Enable STM32Cube DelayBlock (DELAYBLOCK) LL module driver (stm32U5 or STM32N6)
```
```patch
 config USE_STM32_LL_DLYB
 	bool		bool
+	depends on	SOC_SERIES_STM32U5X || SOC_SERIES_STM32N6X
 	help		help
 	  Enable STM32Cube DelayBlock (DELAYBLOCK) LL module driver (stm32U5 or STM32N6)
```
?

erwango replied:
For now we didn't specify these dependencies. If we do, we'd need to review all `USE_STM32_` symbols and I don't think it would scale

etienne-lms reviewed (commented):
Few maybe nitpicking comments...

erwango commented on boards/st/nucleo_n657x0_q/doc/index.rst:
```diff
@@ -0,0 +1,169 @@
+Flashing
+========
+
+The board is configured to be flashed using west `STM32CubeProgrammer`_ runner,
+so its :ref:`installation <stm32cubeprog-flash-host-tools>` is required.
```
Please mention that version v2.18.0 or later is required

gautierg-st replied:
Done

gautierg-st commented on dts/bindings/clock/st,stm32n6-pll-clock.yaml:
```diff
@@ -0,0 +1,62 @@
+  an input frequency from 5 to 50 MHz. PLLM factor is used to set the input
+  clock in this acceptable range.
+
+  Each PLL have one output clock whose frequency can be computed with the
```
Done

gautierg-st commented on drivers/clock_control/clock_stm32_ll_n6.c:
```diff
@@ -0,0 +1,961 @@
+#include <zephyr/arch/cpu.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
```
Done

nordicjm commented on soc/st/stm32/stm32n6x/soc.c:
```diff
@@ -0,0 +1,67 @@
+	return 0;
+}
+
+SYS_INIT(stm32n6_init, PRE_KERNEL_1, 0);
```
should be using soc hook

erwango replied:
This means removing this call and replace `stm32n6_init` with `soc_early_init_hook()`

gautierg-st replied:
Done

erwango commented on boards/st/stm32n6570_dk/doc/index.rst:
```diff
@@ -0,0 +1,176 @@
+=========
+
+For now debugging is only available through STM32CubeIDE:
+* Create an empty STM32 project with your board.
+* Create a default debug configuration.
+* In the debug configuration, disable auto build.
+* In the debug configuration, change the C/C++ Application to your zephyr.elf.
```
Let's update with the new fancy way as this could avoid issues in some cases.

gautierg-st replied:
Done

erwango commented on boards/st/stm32n6570_dk/doc/index.rst:
```diff
@@ -0,0 +1,176 @@
+* Create a default debug configuration.
+* In the debug configuration, disable auto build.
+* In the debug configuration, change the C/C++ Application to your zephyr.elf.
+
```
Also mentions switches position for debug.

gautierg-st replied:
Done

erwango commented on boards/st/stm32n6570_dk/doc/index.rst:
```diff
@@ -0,0 +1,176 @@
+
+Connect the STM32N6570_DK to your host computer using the USB port.
+Then build and flash an application. To Flash, BOOT0 pin should be set to 0 and
+BOOT1 to 1. Once flashed, BOOT1 should be set to 0 to run the application.
```
> To Flash, BOOT0 pin should be set to 0 and
BOOT1 to 1. Once flashed, BOOT1 should be set to 0 to run the application.

I'm affraid this is not outlined enough for "quick" readers.
Can you make it more visible ?

gautierg-st replied:
Done

gautierg-st committed:
soc: st: stm32: add a comment for kconfig source priority

Add a comment describing Kconfig source order priority to prevent
future questions (after overriding SYS_CLOCK_HW_CYCLES_PER_SEC config
for STM32N6).

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

erwango commented on dts/bindings/clock/st,stm32n6-cpu-clock-mux.yaml:
```diff
@@ -0,0 +1,27 @@
+    required: true
+    type: int
+    description: |
+      default frequency in Hz for clock output
```
`      Default frequency in Hz for clock output`. 
Also, be more explicit on the actual clock this represents (sys_cpu_ck)

gautierg-st replied:
Done

erwango commented on dts/bindings/clock/st,stm32n6-cpu-clock-mux.yaml:
```diff
@@ -0,0 +1,27 @@
+
+description: |
+  STM32N6 CPU Clock
+  Describes the STM32N6 CPU clock multiplexer.
```
Let's specifies this is the clock feeding Systick.

gautierg-st replied:
Done

erwango commented on dts/bindings/clock/st,stm32n6-cpu-clock-mux.yaml:
```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2022, Linaro ltd
```
```suggestion
# Copyright (c) 2025, STMicroelectronics
```

gautierg-st replied:
Done

gautierg-st commented on soc/st/stm32/Kconfig.defconfig:
```diff
@@ -18,10 +18,14 @@ config CORTEX_M_SYSTICK
 
 config SYS_CLOCK_HW_CYCLES_PER_SEC
+	default "$(DT_STM32_CPU_CLOCK_FREQ)" if "$(dt_nodelabel_enabled,cpusw)"
 	default "$(DT_STM32_RCC_CLOCK_FREQ)" if "$(dt_nodelabel_enabled,rcc)"
```
Moved the specific N6 definition in N6 Kconfig

gautierg-st commented on dts/bindings/clock/st,stm32n6-cpu-clock-mux.yaml:
```diff
@@ -0,0 +1,27 @@
+  For instance:
+  &perck {
+    clocks = <&rcc STM32_SRC_HSI_KER CKPER_SEL(0)>;
+    status = "okay";
```
Done

gautierg-st commented on dts/bindings/clock/st,stm32n6-cpu-clock-mux.yaml:
```diff
@@ -0,0 +1,27 @@
+  STM32N6 CPU Clock
+  Describes the STM32N6 CPU clock multiplexer.
+  For instance:
+  &perck {
```
Done

kartben commented on boards/st/nucleo_n657x0_q/board.yml:
```diff
@@ -0,0 +1,5 @@
+board:
+  name: nucleo_n657x0_q
```
Missing `full_name`

gautierg-st replied:
Done

kartben commented on boards/st/stm32n6570_dk/board.yml:
```diff
@@ -0,0 +1,5 @@
+board:
+  name: stm32n6570_dk
```
Ditto
